test/suite/test_verify.py

```diff
@@ -167,6 +167,10 @@ def test_verify_api_read_corrupt_pages(self):
         params = 'key_format=S,value_format=S'
         self.session.create('table:' + self.tablename, params)
         self.populate(self.tablename)
+        # Open the file and corrupt the first page. 
+        with self.open_and_position(self.tablename, 0.1) as f:
+            for i in range(0, 100):
+                f.write(b'\x01\xff\x80')
```

Comment:
This will corrupt the first page of the file. But the scenario we are interested in here is when the first block referenced from an internal node has an error. So to test it we need to find and corrupt a specific block of data. Offhand, I think I would do this by running `wt verify -d dump_address` and using the address of the first leaf page it lists.

Reply:
Sorry I may have understood this scenario. From my understanding the scenario where the bugs happens occurs in the following sequence:
1. The first internal page is skipped
2. First child page is corrupted and returns error (This doesn't set the max_addr_size due to failing)
3. When verifying second internal page, we hit the assert because max_addr_size is not set
Let me know what you think?

Reply:
Hi @jiechenbo.

The problem scenario is in [the code](https://github.com/wiredtiger/wiredtiger/blob/a8179abe4b8ed7132ee0f8e47d1286267e9c1c09/src/btree/bt_vrfy.c#L652) in the `WT_PAGE_ROW_INT` case inside `__verify_tree()`. 

We have a problem if the first child is corrupt. (i.e., not the "magic" 0th key mentioned in the comment). In this case `entry` is 1 and we don't call `__verify_row_int_key_order()`. But when that block is corrupt, we discover that in `__wt_page_in()` and we take the `continue` path and don't perform the final work the loop, which would verify the page. We can't verify it because it's corrupt. But something in that verification sets up the state we need when calling `__verify_row_int_key_order()`.

Looking at the issue more carefully, the problem with just disabling the ASSERT is that it assumes that the following code will behave correctly if the ASSERT doesn't hold. Maybe that's true, but I don't understand that functionality well enough to know for sure. I just found the bug and reported it :-) 

I have the data file that produced this issue originally. It's not attached on the ticket since it's 2GB and contains customer data. But I'll send you a pointer so you can reproduce the problem in the debugger and also have something you can test against.

Reply:
>We have a problem if the first child is corrupt. (i.e., not the "magic" 0th key mentioned in the comment). In this case entry is 1 and we don't call __verify_row_int_key_order(). But when that block is corrupt, we discover that in __wt_page_in() and we take the continue path and don't perform the final work the loop, which would verify the page. We can't verify it because it's corrupt. But something in that verification sets up the state we need when calling __verify_row_int_key_order().
Hmm interesting, sorry I may have misworded the comment in this place. The test scenario plans to corrupt multiple blocks inside the first page, in which we call verify after which will reproduce this case.

> Looking at the issue more carefully, the problem with just disabling the ASSERT is that it assumes that the following code will behave correctly if the ASSERT doesn't hold. Maybe that's true, but I don't understand that functionality well enough to know for sure. I just found the bug and reported it :-)

Yes the `__verify_row_int_key_order` will have update the max_addr_size and max_key if we disable it here.

Reply:
Thank you for the well written explanation! I have added a case here [c219edd](https://github.com/wiredtiger/wiredtiger/pull/9209/commits/c219edde9febc9134dc408bae2b11481c3d07180) , let me know what you think